java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Appetizer/AppetizerRepository.java

```diff
@@ -1,16 +1,68 @@
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Repository;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.server.ResponseStatusException;
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
+
 @Repository
+@RequiredArgsConstructor
+@Slf4j
 public class AppetizerRepository {
-    private final FoodTruck2 appetizer1 = new FoodTruck2(1,"Garlic bread","18/22","21/22",3);
-    private final FoodTruck2 appetizer2 = new FoodTruck2(2,"Breadsticks ","11/22","28/22",2);
-    ArrayList<FoodTruck2> appetizers = new ArrayList<>(Arrays.asList(appetizer1,appetizer2));
+   private final JdbcTemplate jdbcTemplate;
 
-    public ArrayList<FoodTruck2> getAllAppetizers(){
-        return appetizers;
+    //   ----------- INSERTING A NEW APPETIZER -----------  //
+    public Appetizer creatingNewAppetizer(@RequestBody AppetizerRequestBody appetizerRequestBody){
+        String sql = "INSERT INTO appetizer(name,date,expire_date,price) VALUES(?,?,?,?) RETURNING *";
+        Appetizer appetizer = jdbcTemplate.queryForObject(sql,new BeanPropertyRowMapper<>(Appetizer.class),
+                appetizerRequestBody.getAppetizerName(),
+                appetizerRequestBody.getDate(),
+                appetizerRequestBody.getExpireDate(),
+                appetizerRequestBody.getPrice());
+        log.info("new appetizer AVAILABLE!!!! ");
+        return appetizer;
+    }
+    // ---------GETTING APPETIZER BY THEIR ID----------//
+    public Appetizer gettingAppetizerById(@PathVariable Integer id){
+        String sql = "SELECT * FROM appetizer WHERE id = ?";
+        try {
+            Appetizer appetizer = jdbcTemplate.queryForObject(sql,new BeanPropertyRowMapper<>(Appetizer.class),id);
+            log.info("Found the appetizer with the id: " + id);
+                return appetizer;
+        }catch (EmptyResultDataAccessException emptyResultDataAccessException){
+            log.error("This id: " +id + " does NOT EXIST");
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+    }
+    // -----------GETTING ALL APPETIZER------------ //
+    public List<Appetizer> gettingALlAppetizer(){
+        String sql = "Select * FROM appetizer";
+        List<Appetizer> appetizerList = jdbcTemplate.query(sql,new BeanPropertyRowMapper<>(Appetizer.class));
+        return appetizerList;
+    }
+    // ----------- DELETING CUSTOMERS BY THEIR ID -------------//
+    public void deleteAppetizerById(@PathVariable Integer id ){
+        String sql = "SELECT * FROM appetizer WHERE id = ?";
+        String sqlDelete = "DELETE FROM appetizer WHERE id = ?";
+        try {
+            jdbcTemplate.queryForObject(sql,new BeanPropertyRowMapper<>(Appetizer.class),id);
+            log.info("Appetizer has been Deleted!! USER ID: " + id);
```

Comment:
line 55 (comment) and the log on line 61 are still incorrect. This is related to an appetizer. Not a customer...
I can see you just copied + pasted here

Reply:
I only Copied the line 55, the rest was my own mistakes, next time i read my code out load to myself to see i am not writing wrong